language/go/fix.go

```diff
@@ -72,6 +177,9 @@ func migrateGrpcCompilers(c *config.Config, f *rule.File) {
 // MergeFile will remove unused values and attributes later.
 func squashCgoLibrary(c *config.Config, f *rule.File) {
 	// Find the default cgo_library and go_library rules.
+	binName := binName(f)
```

Comment:
No change needed to any of these fixes. They just need to run first.

Reply:
Which tests fail exactly? I'm assuming just stuff in fix_test.go.

If the old tests are failing, I think it's fine to lock those into the old naming convention.

If the new tests are failing, some change may be needed. The fixes are transformations run in sequence. Earlier transformations should work without knowledge of newer changes, but maybe they just need to be fixed to ignore files already written with the new naming convention?

Reply:
What's the harm though in updating these?

Reply:
It adds complexity and requires additional testing.

The `Fix` method is meant to be like `go fix`. In the early days of Gazelle and rules_go, there were some conventions that we wanted to migrate away from. For example, a separate `cgo_library` target was needed for cgo code, and separate `go_test` targets were needed for internal and external test packages.

Gazelle won't generate targets in those conventions anymore, but we wanted to help people migrate away from them automatically. So `Fix` applies a stack of transformations. 

For the most part, these transformations interact very little, but the stack is loosely ordered, so a transformation may assume that if it needs to be applied, then later transformations haven't been applied yet. For example, `squashCgoLibrary` doesn't need to worry about naming conventions since `migrateNamingConvention` will still be applied later.

Ideally, we should be able to add a new transformation on this stack without modifying any existing transformation. Transformations shouldn't need to support being applied in an arbitrary order.